http-client/src/test/groovy/io/micronaut/http/client/format/ClientFormatSpec.groovy

```diff
@@ -0,0 +1,176 @@
+package io.micronaut.http.client.format
+
+import io.micronaut.context.annotation.Property
+import io.micronaut.context.annotation.Requires
+import io.micronaut.core.annotation.Introspected
+import io.micronaut.core.beans.exceptions.IntrospectionException
+import io.micronaut.core.convert.format.Format
+import io.micronaut.http.HttpRequest
+import io.micronaut.http.annotation.Controller
+import io.micronaut.http.annotation.Get
+import io.micronaut.http.annotation.QueryValue
+import io.micronaut.http.client.annotation.Client
+import io.micronaut.test.extensions.spock.annotation.MicronautTest
+import jakarta.inject.Inject
+import spock.lang.Specification
+
+@Property(name = 'spec.name', value = 'ClientFormatSpec')
+@MicronautTest
+class ClientFormatSpec extends Specification {
+    @Inject FormatClient client
+
+    void "test PIPES formatted list"() {
+        given:
+        var values = ["tree", "spruce", "seed"]
+        expect:
+        client.pipesFormattedList(values) == "param=tree|spruce|seed"
+    }
+
+    void "test PIPES formatted Map"() {
+        given:
+        var values = ["name": "Mark", "age": 30]
+        expect:
+        client.pipesFormattedMap(values) == "param=name|Mark|age|30"
+    }
+
+    void "test PIPES formatted Object"() {
+        given:
+        var cafe = new Cafe(name: "Pizza Garden", "address": "Home Street 2")
+        expect:
+        client.pipesFormattedObject(cafe) == "param=name|Pizza+Garden|address|Home+Street+2"
+    }
+
+    void "test csv format all values"() {
+        expect:
+        client.csvFormattedObject(param) == query
+
+        where:
+        param                                  | query
+        ["a", "b", "c"]                        | "csv=a,b,c"
+        ["name": "Bob", "age": 12]             | "csv=name,Bob,age,12"
+        new Cafe(name: "AAA", address: "here") | "csv=name,AAA,address,here"
```

Comment:
These use cases don't make sense to me. It seems really odd to want to convert a map to a csv where the keys of the map are part of the values. Do you have some reference behavior in some other framework or documentation that was used as the inspiration?

Reply:
It is here if you scroll to the Query parameters and see the "form" style in the table: [Open api v3 parameter serialization](https://swagger.io/docs/specification/serialization/).
But I do agree that it doesn't make much sense.

Reply:
In that table it shows N/A for `spaceDelimited` and `pipeDelimited` with objects

Reply:
It may just make sense to support it though to be consistent. Seems odd its N/A on their side